src/videojs-http-streaming.js

```diff
@@ -795,7 +795,19 @@ class VhsHandler extends Component {
       },
       bandwidth: {
         get() {
-          return this.masterPlaylistController_.mainSegmentLoader_.bandwidth;
+          let bandwidthEst = this.masterPlaylistController_.mainSegmentLoader_.bandwidth;
+
+          // NetworkInfo.downlink maxes out at 10 Mbps. In the event that the player estimates a bandwidth
+          // greater than 10 Mbps, use the larger value to ensure that high quality streams are not
+          // accidentally filtered out
+          if (this.options_.useNetworkInformationApi && this.networkInformation) {
+            // Downlink property returns Mbps https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/downlink
+            const effectiveBandwidthBitsPerSec = this.networkInformation.downlink * 1000 * 1000;
+
+            bandwidthEst = Math.max(bandwidthEst, effectiveBandwidthBitsPerSec);
+          }
```

Comment:
I think we can remove the `this.networkInformation` property and create a variable for it here to avoid creating another private property and keep everything in one spot.

```suggestion
          const  networkInformation = window.navigator.connection || window.navigator.mozConnection || window.navigator.webkitConnection;

          if (this.options_.useNetworkInformationApi && networkInformation) {
            // downlink returns Mbps
            // https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/downlink
            const effectiveBandwidthBitsPerSec = this.networkInformation.downlink * 1000 * 1000;

            // downlink maxes out at 10 Mbps. In the event that the player estimates a bandwidth
            // greater than 10 Mbps, use the larger value to ensure that high quality streams are not
            // filtered out.
            bandwidthEst = Math.max(bandwidthEst, effectiveBandwidthBitsPerSec);
          }
```